vector/src/main/java/im/vector/riotx/features/home/room/detail/timeline/action/ViewEditHistoryViewModel.kt

```diff
@@ -79,16 +82,36 @@ class ViewEditHistoryViewModel @AssistedInject constructor(@Assisted
             }
 
             override fun onSuccess(data: List<Event>) {
-                //TODO until supported by API Add original event manually
-                val withOriginal = data.toMutableList()
                 var originalIsReply = false
-                room.getTimeLineEvent(eventId)?.let {
-                    withOriginal.add(it.root)
-                    originalIsReply = it.root.getClearContent().toModel<MessageContent>().isReply()
+
+                val events = data.map {
+                    val timelineID = it.roomId + UUID.randomUUID().toString()
+                    it.apply {
+                        //We need to check encryption
+                        if (isEncrypted() && mxDecryptionResult == null) {
+                            //for now decrypt sync
+                            try {
+                                val result = session.decryptEvent(this, timelineID)
+                                mxDecryptionResult = OlmDecryptionResult(
+                                        payload = result.clearEvent,
+                                        senderKey = result.senderCurve25519Key,
+                                        keysClaimed = result.claimedEd25519Key?.let { k -> mapOf("ed25519" to k) },
+                                        forwardingCurve25519KeyChain = result.forwardingCurve25519KeyChain
+                                )
+                            } catch (e: MXCryptoError) {
+
+                            }
+                        }
+
+                        if (it.eventId == eventId) {
```

Comment:
I'm not comfortable with the `apply` here, because, there is 3 different `eventId` in the class...